# **Maiores Sálarios da Área de Ciência de Dados**

Banco de Dados: *Jobs and Salaries in Data Science*

Fonte: Kaggle, acesso: https://www.kaggle.com/datasets/hummaamqaasim/jobs-in-data/data


# **Descrição do Banco de Dados**

Esse conjunto de dados reune informações reais sobre empregos na área de ciência de dados. Aqui são feitas análises de diversos pontos ao redor da faixa salarial e da experiência do colaborador da área.

## Tipo de Informação

- Salários
- Cargos
- Áreas
- Experiência do colaborador
- Tamanho da empresa
- Modelo de trabalho
- Período de Trabalho
- País de Residência

## **Discussão**

A análise desse banco de dados é relevante para responder questionamentos relacionados a salários, benefícios, tempo de dedicação e oportunidades dentro da Área de dados ao redor do mundo.

# **O que é um DataFrame?**


*Explicação*: DataFrame é uma tabela com linhas e colunas, como um excel, porém é bastante utilizada para manipulação e análise de dados através da programação em python ou de uma forma que seja sua preferência, assim como a do mercado de trabalho!

*Estrutura de dados tabular*:

# **Bibliotecas**

Pandas: É a principal ferramenta para a análise de Dados, com ela é possível manipular e analisar todas as informações presentes em um DataFrame.

Numpy: É responsável pelas operações numéricas que podem ser necessárias durante os calcúlos exigidos durante a inspeção e verificação de informações.

Matplotlib: É a biblioteca utilizada para mostrar o resultado dos dados analisados, com ela é possível montar gráficos de diferentes tipos.

In [4]:
# Bibliotecas
from google.colab import drive
drive.mount('/content/drive')

import pandas as pd
import numpy as np

import matplotlib.pyplot as plt

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [5]:
df = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/jobs_in_data.csv')

## **Discussão**

- from google.colab import drive - Permite puxar o DataFrame do drive, para que não seja necessário conter o arquivo no computador, somente no drive.

- drive.mount('/content/drive') - Indica o caminho do arquivo no drive

- import panda as pd - Importa a biblioteca pandas e a nomeia como pd, facilitando a utilização da mesma no código.

- import numpy as np - Importa a biblioteca numpy e a nomeia como np, facilitando a utilização da mesma no código.

- import matplotlib.pyplot as plt -Importa a biblioteca matplotlib e a nomeia como plt, facilitando a utilização da mesma no código.

- df = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/jobs_in_data.csv') - Trazemos o DataFrame para o python, permitindo os próximos passos na análise.


# **Visualização Inicial**

- Para a nossa análise inicial, podemos utilizar o comando: display(df.head(n)). Com ele é possível ver as primeiras linhas do DataFrame!

- (n) - Significa a quantidade de linhas que você quiser ver, começando das linhas iniciais da tabela.

- Outro comando relevante é: display(df.tail(n)). Ele funciona assim como o comando anterior, porém ele mostra as últimas linhas do DataFrame.

In [6]:
#print(df.head(10))
display(df.head(10))

,work_year,job_title,job_category,salary_currency,salary,salary_in_usd,employee_residence,experience_level,employment_type,work_setting,company_location,company_size
0,2023,Data DevOps Engineer,Data Engineering,EUR,88000,95012,Germany,Mid-level,Full-time,Hybrid,Germany,L
1,2023,Data Architect,Data Architecture and Modeling,USD,186000,186000,United States,Senior,Full-time,In-person,United States,M
2,2023,Data Architect,Data Architecture and Modeling,USD,81800,81800,United States,Senior,Full-time,In-person,United States,M
3,2023,Data Scientist,Data Science and Research,USD,212000,212000,United States,Senior,Full-time,In-person,United States,M
4,2023,Data Scientist,Data Science and Research,USD,93300,93300,United States,Senior,Full-time,In-person,United States,M
5,2023,Data Scientist,Data Science and Research,USD,130000,130000,United States,Senior,Full-time,Remote,United States,M
6,2023,Data Scientist,Data Science and Research,USD,100000,100000,United States,Senior,Full-time,Remote,United States,M
7,2023,Machine Learning Researcher,Machine Learning and AI,USD,224400,224400,United States,Mid-level,Full-time,In-person,United States,M
8,2023,Machine Learning Researcher,Machine Learning and AI,USD,138700,138700,United States,Mid-level,Full-time,In-person,United States,M
9,2023,Data Engineer,Data Engineering,USD,210000,210000,United States,Executive,Full-time,Remote,United States,M


In [7]:
display(df.tail(10))

,work_year,job_title,job_category,salary_currency,salary,salary_in_usd,employee_residence,experience_level,employment_type,work_setting,company_location,company_size
9345,2020,Machine Learning Engineer,Machine Learning and AI,EUR,40000,45618,Croatia,Senior,Full-time,Remote,Croatia,S
9346,2021,Director of Data Science,Data Science and Research,USD,168000,168000,Japan,Senior,Full-time,In-person,Japan,S
9347,2021,Data Scientist,Data Science and Research,SGD,160000,119059,Singapore,Mid-level,Full-time,Remote,Israel,M
9348,2021,Applied Machine Learning Scientist,Machine Learning and AI,USD,423000,423000,United States,Mid-level,Full-time,Hybrid,United States,L
9349,2021,Data Engineer,Data Engineering,EUR,24000,28369,Malta,Mid-level,Full-time,Hybrid,Malta,L
9350,2021,Data Specialist,Data Management and Strategy,USD,165000,165000,United States,Senior,Full-time,Remote,United States,L
9351,2020,Data Scientist,Data Science and Research,USD,412000,412000,United States,Senior,Full-time,Remote,United States,L
9352,2021,Principal Data Scientist,Data Science and Research,USD,151000,151000,United States,Mid-level,Full-time,Remote,United States,L
9353,2020,Data Scientist,Data Science and Research,USD,105000,105000,United States,Entry-level,Full-time,Remote,United States,S
9354,2020,Business Data Analyst,Data Analysis,USD,100000,100000,United States,Entry-level,Contract,Remote,United States,L


**Importância**: A análise inicial é importante para identificar informações sobre o banco de Dados, assim como iniciar o seu processo de limpeza.

# **Variáveis**

Tipos de variáveis:

- Qualitativas -
  - Categóricas: (Nominais e ordinais) -
- Quantitativas -
  - Numéricas: (Discretas e contínuas) -

**Variáveis do DataFrame**:
- work_year - Numérica
- job_title - Categórica
- job_category - Categórica
- salary_currency - Categórica
- salary - Numérica
- salary_in_usd - Numérica
- employee_residence - Categórica
- experience_level - Categórica
- employment_type - Categórica
- work_setting - Categórica
- company_location - Categórica
- company_size - Categórica


df.info() - Traz o nome das colunas, o tipo de dados que são armazenados nelas, a quantidade de entradas no banco, quantidade de colunas e memória que o banco ocupa.

In [10]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9355 entries, 0 to 9354
Data columns (total 12 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   work_year           9355 non-null   int64 
 1   job_title           9355 non-null   object
 2   job_category        9355 non-null   object
 3   salary_currency     9355 non-null   object
 4   salary              9355 non-null   int64 
 5   salary_in_usd       9355 non-null   int64 
 6   employee_residence  9355 non-null   object
 7   experience_level    9355 non-null   object
 8   employment_type     9355 non-null   object
 9   work_setting        9355 non-null   object
 10  company_location    9355 non-null   object
 11  company_size        9355 non-null   object
dtypes: int64(3), object(9)
memory usage: 877.2+ KB


dt.types - Uma outra forma de analisar nosso DataFrame, aqui o foco são nossas variáveis e os tipos delas para a análise!

In [12]:
df.dtypes

,0
work_year,int64
job_title,object
job_category,object
salary_currency,object
salary,int64
salary_in_usd,int64
employee_residence,object
experience_level,object
employment_type,object
work_setting,object


## **Discussão**

**Importância**: É necessário analisar o tipo da variável, para que possamos definir a forma de transformar os dados do tipo daquela coluna, como também conseguir entender como transformar as informações em uma representação visual

**Significado de cada Variável do Banco de Dados:**
- work_year - Ano do trabalho (Que as informações foram coletadas)
- job_title - Cargo do trabalho
- job_category - Área do trabalho dentro de Dados
- salary_currency - Moeda
- salary - Salário da moeda original
- salary_in_usd - Salário convertido em doláres
- employee_residence - País de residência do colaborador
- experience_level - Nível de Expêriencia
- employment_type - Período de trabalho (Integral, matutino e etc)
- work_setting - Tipo de trabalho (Remoto, presencial e etc)
- company_location - País da empresa do colaborador
- company_size - Tamanho da empresa do colaborador

# **Verificações**

Explicação: Durante a Análise exploratória de Dados (EDA), é necessário verificar se existem informações nulas ou duplicadas no DataFrame. Essa etapa é essencial porque os dados que forem obtidos com ignorância dessa parte pode afetar o resultado final de uma forma extremamente prejudicial.

Portanto, vamos verificar os espaços Vazios/Valores nulos por colunas a seguir!

In [13]:
df.isnull().sum()

,0
work_year,0
job_title,0
job_category,0
salary_currency,0
salary,0
salary_in_usd,0
employee_residence,0
experience_level,0
employment_type,0
work_setting,0


- *df.isnull()* - Procura por valores nulos no DataFrame
- *.sum()* - Faz a contagem de todos os valores procurados, para poder mensurar a quantidade!

**Observação**: Como o valor foi 0, isso significa que não existem valores nulos/vazios nesse DataFrame, todas as linhas de todas as colunas foram preenchidas!

In [15]:
df.duplicated().sum()

np.int64(4014)

- df.duplicated().sum() - Esse comando procura por linhas duplicadas e contabiliza elas, portanto temos 4014 linhas duplicadas aqui.

In [16]:
df[df.duplicated()]

,work_year,job_title,job_category,salary_currency,salary,salary_in_usd,employee_residence,experience_level,employment_type,work_setting,company_location,company_size
64,2023,Data Analyst,Data Analysis,USD,75000,75000,United States,Entry-level,Full-time,In-person,United States,M
111,2023,Research Scientist,Data Science and Research,USD,165000,165000,United States,Mid-level,Full-time,Remote,United States,M
112,2023,Research Scientist,Data Science and Research,USD,118800,118800,United States,Mid-level,Full-time,Remote,United States,M
115,2023,Data Engineer,Data Engineering,USD,184000,184000,United States,Mid-level,Full-time,In-person,United States,M
116,2023,Data Engineer,Data Engineering,USD,123000,123000,United States,Mid-level,Full-time,In-person,United States,M
...,...,...,...,...,...,...,...,...,...,...,...,...
9079,2022,Data Scientist,Data Science and Research,USD,78000,78000,United States,Mid-level,Full-time,Remote,United States,M
9080,2022,Data Engineer,Data Engineering,USD,135000,135000,United States,Senior,Full-time,Remote,United States,M
9081,2022,Data Engineer,Data Engineering,USD,115000,115000,United States,Senior,Full-time,Remote,United States,M
9214,2021,Data Engineer,Data Engineering,USD,200000,200000,United States,Mid-level,Full-time,Remote,United States,L


- **df[df.duplicated()]** - Traz todas as linhas que são duplicadas no DataFrame, sendo possível analisar todas as informações delas.

In [22]:
df_reduzido = df.drop_duplicates().copy()
df_reduzido.duplicated().sum()

**Explicação**:
- df_reduzido = df.drop_duplicates().copy() - Aqui é feita uma cópia do DataFrame original, essa cópia exclui todas as linhas duplicadas do Banco.
- df_reduzido.duplicated().sum() - Para garantir que a nossa cópia (Feita para não gerar problemas no Banco original) cumpriu a função, verificamos a quantidade de linhas duplicadas. (Como retornou 0, isso significa que não existem mais linhas duplicadas no nosso novo DataFrame)

In [21]:
df.T.duplicated().sum()

np.int64(0)

## **Observações da Análise Inicial**:
- df.T.duplicated().sum() - Verificação das colunas duplicadas no DataFrame original
- Não existem colunas duplicadas no Dataframe
- Não existem valores nulos no Banco de Dados
- Antes da limpeza, 4014 linhas eram duplicadas, agora não existem mais linhas duplicadas.
- Duplicação de linhas X Duplicação de colunas - Quando uma linha é duplicada, temos dados presentes em todas as colunas que são duplicados. A duplicação delas é complicada de lidar, porque ela pode atrapalhar o resultado final da análise, com valores e informações que geram desvios incorretos.

  Quando uma coluna é duplicada, isso significa que ocorre uma redundância do Banco, porque é o mesmo dado em um frequência desnecessária e prejudicial. A duplicação de colunas ocorre quando existe algum erro de código ou caso seja as mesmas informações armazenadas em duas colunas que são também, a mesma coisa.
- Dados duplicados afetam médias, contagens e conclusões, cada valor pode gerar uma alteração que pode ser pequena, mas que gera diferenças do real que podem atrapalhar dependendo do tamanho do banco, sendo uma solução de tamanho empresarial por exemplo.

# **Estatísticas Descritivas**

In [9]:
df.describe()

,work_year,salary,salary_in_usd
count,9355.000000,9355.000000,9355.000000
mean,2022.760449,149927.981293,150299.495564
std,0.519470,63608.835387,63177.372024
min,2020.000000,14000.000000,15000.000000
25%,2023.000000,105200.000000,105700.000000
50%,2023.000000,143860.000000,143000.000000
75%,2023.000000,187000.000000,186723.000000
max,2023.000000,450000.000000,450000.000000


**Explicação**: df.describe() - Mostra informações matemáticas e estatísticas relacionadas as colunas numéricas do Banco de Dados.

## **Discussão**

As estatísticas tragas são fundamentais para pontuar diferenciais do DataFrame.

Informações como: Média, moda, mediana e um boxplot, podendo adicionar na procura de outliers (Pontos fora da curva).

Nessa etapa, é importante analisar os dados que limpamos para achar outras formas de eliminar as redundâncias ou outros fatores sem sentido do Banco.

# **Padronização**